lib/core/src/persist/mod.rs

```diff
@@ -85,16 +86,33 @@ impl Persister {
         network: LiquidNetwork,
         sync_enabled: bool,
         asset_metadata: Option<Vec<AssetMetadata>>,
+        backup_bytes: Option<Vec<u8>>,
     ) -> Result<Self> {
         let main_db_dir = PathBuf::from_str(database_id)?;
-        Self::new_inner(main_db_dir, network, sync_enabled, asset_metadata)
+        let backup_con = backup_bytes
+            .map(|data| {
+                let size = data.len();
+                let cursor = std::io::Cursor::new(data);
+                let mut conn = Connection::open_in_memory()?;
+                conn.deserialize_read_exact(rusqlite::DatabaseName::Main, cursor, size, false)?;
```

Comment:
I wonder what we should do in the case the restore of the backup fails (maybe the file/indexedDb get corrupted). Maybe if restore fails we should continue without and re-hydrate via data-sync/recoverer. If we do nothing the SDK is basically bricked until the web user clears the indexedDB table or nodejs user deletes/changes the workingDir

Reply:
Good point! Done in [2277efc](https://github.com/breez/breez-sdk-liquid/pull/851/commits/2277efcd81036f9335bd79d211001252f1f307c1)